poppy/poppy_core.py

```diff
@@ -119,29 +120,30 @@ class Wavefront(object):
 
     """
 
-    def __init__(self,wavelength=2e-6, npix=1024, dtype=np.complex128, diam=8.0, oversample=2, pixelscale=None):
+    @quantity_input(wavelength=u.meter, diam=u.meter, pixelscale=u.arcsec/u.pixel)
+    def __init__(self,wavelength=2e-6*u.meter, npix=1024, dtype=np.complex128, diam=8.0*u.meter, oversample=2, pixelscale=None):
 
-        if wavelength > 1e-4:
-            raise ValueError("The specified wavelength %f is implausibly large. Remember to specify the desired wavelength in *meters*." % wavelength)
+        #if wavelength > 1e-3*u.m:
```

Comment:
I guess this is how POPPY becomes a general-purpose physical optics library? 😉 More commented code to be removed